base/loading.jl

```diff
@@ -2249,6 +2249,9 @@ end
 # this is called in the external process that generates precompiled package files
 function include_package_for_output(pkg::PkgId, input::String, depot_path::Vector{String}, dl_load_path::Vector{String}, load_path::Vector{String},
                                     concrete_deps::typeof(_concrete_dependencies), source::Union{Nothing,String})
+    if !occursin("module ", readuntil(input, "module ", keep = true))
+        throw(ErrorException("Package $pkg source file $input does not contain a module."))
+    end
```

Comment:
Perhaps a more robust solution that respects comments and allows for a macro
```julia
rgx = r"^\s*(?:@\w*\s*)module"m
if occursin(input, rgx)
  throw(ErrorException("Package $pkg source file $input does not contain a module."))
end
```

Reply:
Input is a path and there's no `readuntil` that takes a regex. Also it should accept baremodule I think.

Reply:
Upsi.
This here would work with a filename
```julia
    rgx = r"^\s*(?:@\w*\s*)*(?:bare)?module"
    if !any(l -> contains(l, rgx), eachline(input))
        # throw
    end
```

Reply:
Added, thanks!